templates/home.html

```diff
@@ -32,12 +32,23 @@ <h4>All Tweets</h4>
   {% for tweet in all_tweets %}
   <div class="tweet">
     <h6>{{ tweet.user.first_name }}</h6>
-    <p>{{ tweet.tweet_message }}</p>
+    {% if tweet.original_tweet is None %}
+      <p>{{ tweet.tweet_message }}</p>
+    {% else %}
+      <p>Retweeted</p>
+      <h6>{{ tweet.original_tweet.user.first_name }}</h6>
+      <p>{{ tweet.original_tweet.tweet_message }}</p>
+    {% endif %}
     <a href="{% url 'tweets:showTweet' tweet.id %}">Show Tweet</a>
-    {% if request.user.id ==  tweet.user_id %}
+    {% if request.user.id ==  tweet.user_id and tweet.original_tweet is None %}
       <a href="{% url 'tweets:editTweet' tweet.id %}" style="margin-left: 20px;">Edit Tweet</a>
+    {% endif %}
+    {% if request.user.id ==  tweet.user_id %}
       <a href="{% url 'tweets:destroyTweet' tweet.id %}" style="float: right">Delete Tweet</a>
     {% endif %}
+    {% if request.user.id !=  tweet.user_id and request.user.is_authenticated and tweet.original_tweet is None %}
+      <a href="{% url 'tweets:reTweet' tweet.id %}" style="margin-left: 20px;">ReTweet</a>
+    {% endif %}
```

Comment:
Too many if else to understand to understand. Please use some helper methods. e.g Add instance method  to tweet Model 

```
not_a_retweet?
   original_tweet is None
```

or 

```
is_an_orginal_tweet?
   original_tweet is None
```

Reply:
okay. hepler functions are made for views. for direct html we can use templatetags. should I do that?

Reply:
```python
not_a_retweet?
   original_tweet is None
``` 
also this can be done direct with templatetags I think something like -
```python
{% if can_retweet request.user tweet %}
    <a href="{% url 'tweets:reTweet' tweet.id %}" style="margin-left: 20px;">ReTweet</a>
{% endif %}
``` 
but there will be 3 like the above for edit and delete also. But yes code can be more easily readable.